Dockerfile

```diff
@@ -47,7 +47,7 @@ COPY --chown=nonroot . .
 # Show build caching stats to check if it was used in the end.
 # Has to be the part of the same RUN since cachepot daemon is killed in the end of this RUN, losing the compilation stats.
 RUN set -e \
-    && mold -run cargo build  \
+    && RUSTFLAGS="-Clinker=clang -Clink-arg=-fuse-ld=mold -Clink-arg=-Wl,--no-rosegment" cargo build  \
```

Comment:
Should we set the same flags for our test infrastructure as well (maybe for `release` build_type only)?

https://github.com/neondatabase/neon/blob/7d3cdc05d486ee1a1ef5ec8d7137949bcf7d036e/.github/workflows/build_and_test.yml#L357-L359

Reply:
I don't think these specific linker args will influence the results, so I don't think it's necessary.

Thoughts, @koivunej ?

If we do it, some notes:

* The `cov_prefix` script (`scripts/coverage`) also messes with RUSTFLAGS.
* It has logic to merge existing env var, so, our RUSTFLAGS would have to come in front.
* Don't know how it will affect coverage results.

Reply:
FYI, we collect code coverage only for debug builds, so if we change the build command only for release build type it won't affect the coverage results.

Reply:
The very little docs I can find for this would suggest that instead of many segments you get one segment, android and chromium seem to use it. Probably the worst outcome is slight perf hit for mold, but I don't understand why cannot the unwinders be fixed.

So my analysis is that these wont change anything only allow unwinders to search less.

Reply:
Yep, my assessment as well. Dismissing this conversation for now, but, longer term, we should find a solution to centralize RUSTFLAGS

Reply:
https://github.com/neondatabase/neon/issues/6782